server/docker-entrypoint.sh

```diff
@@ -2,29 +2,152 @@
+ceiling() {
+  awk -vnumber="$1" -vdiv="$2" '
+    function ceiling(x){
+      return x%1 ? int(x)+1 : x
+    }
+    BEGIN{
+      print ceiling(number/div)
+    }
+  '
+}
+
+core_limit() {
+  local cpu_period_file="/sys/fs/cgroup/cpu/cpu.cfs_period_us"
+  local cpu_quota_file="/sys/fs/cgroup/cpu/cpu.cfs_quota_us"
+  if [ -r "${cpu_period_file}" ]; then
+    local cpu_period="$(cat ${cpu_period_file})"
+
+    if [ -r "${cpu_quota_file}" ]; then
+      local cpu_quota="$(cat ${cpu_quota_file})"
+      # cfs_quota_us == -1 --> no restrictions
+      if [ "x$cpu_quota" != "x-1" ]; then
+        ceiling "$cpu_quota" "$cpu_period"
+      fi
+    fi
+  fi
+}
+
+max_memory() {
+  local max_mem_unbounded="562949953421312"
+  local mem_file="/sys/fs/cgroup/memory/memory.limit_in_bytes"
+  if [ -r "${mem_file}" ]; then
+    local max_mem="$(cat ${mem_file})"
+    if [ ${max_mem} -lt ${max_mem_unbounded} ]; then
+      echo "${max_mem}"
+    fi
+  fi
+}
+
 BIND=$(hostname -i)
 SERVER=/opt/jboss/infinispan-server
 LAUNCHER=$SERVER/bin/standalone.sh
 CONFIG=clustered
 BIND_OPTS="-Djboss.bind.address.management=0.0.0.0 -Djgroups.join_timeout=1000 -Djgroups.bind_addr=$BIND -Djboss.bind.address=$BIND"
+RUN_TYPE='STANDALONE'
+CONTAINER_SETTINGS="true"
+SERVER_OPTIONS=""
+# Xms should always be set: https://developers.redhat.com/blog/2014/07/15/dude-wheres-my-paas-memory-tuning-javas-footprint-in-openshift-part-1/
+JAVA_OPTS="-Xms64m -Djava.net.preferIPv4Stack=true"
+PERCENT_OF_MEMORY_FOR_MX=70
+
+for i in "$@"
+do
+case $i in
+    domain-controller)
+    RUN_TYPE='DOMAIN_CONTROLLER'
+    shift
+    ;;
+    host-controller)
+    RUN_TYPE='HOST_CONTROLLER'
+    shift
+    ;;
+    -h|--help)
+    echo "######################################################################################"
+    echo "# This script is responsible for setting basic                                       #"
+    echo "# configuration for running Infinispan Server in Docker                              #"
+    echo "#                                                                                    #"
+    echo "# Parameters:                                                                        #"
+    echo "# docker-entrypoint.sh domain-controller [other options]                             #"
+    echo "#     Creates managment user and starts a master controller process                  #"
+    echo "# docker-entrypoint.sh host-controller [other options]                               #"
+    echo "#     Starts a slave instance and connects to a domain controller                    #"
+    echo "# docker-entrypoint.sh host-controller [-n|--no-container-settings] [other options]  #"
+    echo "#     Starts a default standalone Server                                             #"
+    echo "#     -n|--no-container-settings omits memory and CPU settings for container mode    #"
```

Comment:
Exactly! Please note that this change is mainly designed for [setting limits in OpenShift/Kube](https://kubernetes.io/docs/admin/limitrange). I would like to have a safety strip to disable auto-scaling Xmx/CPU functionality in case of any bugs etc.